src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -2211,7 +2211,12 @@ private void ThrowEventSourceException(string eventName, Exception innerEx = nul
                         break;
                     default:
                         if (innerEx != null)
+                        {
+                            // The innermost exception is the interesting one.  
+                            while (innerEx.InnerException != null)
+                                innerEx = innerEx.InnerException;
```

Comment:
@brianrob 

This change is actually unrelated to the Thread Name Setting and I accidentally included it in this PR.   I am thinking however to just include it as it is a useful change that is very low risk.   Basically the issue is that we have had a report from the field where people made mistakes in their EventSource and we logged a nice EventSourceMessage about what they did wrong, but the message they got was 'TypeInitializationError' because the exception happened in the class constructor so it got 'wrapped' in a TypeInitializationError.     Here i am simply unwrapping to get the inner exception (which is almost certainly the interesting thing).     Alternatively we could print out the more verbose exception 'ToString()' which has that and more.    I actually suspect that this is nicer (because most frequently it will be a EventSource-specific exception).    If you want to handle it differently (or as its own PR), just let me know.

Reply:
You could use:
```C#
innerEx = innerEx.GetBaseException();
```
instead.